cdap-ui/app/cdap/components/DataPrep/store/index.js

```diff
@@ -159,10 +165,29 @@ const error = (state = errorInitialState, action = defaultAction) => {
   return Object.assign({}, state, stateCopy);
 };
 
+const workspaces = (state = workspacesInitialState, action = defaultAction) => {
+  let stateCopy;
+
+  switch (action.type) {
+    case DataPrepActions.setWorkspaceList:
+      stateCopy = Object.assign({}, state, {
+        list: action.payload.list
+      });
+      break;
+    case DataPrepActions.reset:
+      return workspacesInitialState;
+    default:
+      return state;
+  }
+
+  return Object.assign({}, state, stateCopy);
+};
+
 const DataPrepStore = createStore(
   combineReducers({
     dataprep,
-    error
+    error,
+    workspaces
   }),
   defaultInitialState,
```

Comment:
Noticed this in the dataprep store too but initial state for the store should be

```
createStore(
  combineReducers({
    dataprep,
    error,
    workspaces
  }),
  {
    dataprep: defaultInitialState,
    error: errorInitialState,
    workspaces: workspacesInitialState
  }
)
```

This is the initial state of the store that has those properties. A similar case was failing in Dataprep Store's Unit tests. Not sure if I am missing something